### PR TITLE
feat(network): add board change

### PR DESCRIPTION
**Pull request by LeonVreling**
Implementes [MA-140](https://myaegee.atlassian.net/browse/MA-140) (hopefully)

[MA-140]: https://myaegee.atlassian.net/browse/MA-140?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by LeonVreling** [comment]
Backend added for adding new boards

**Comment by LeonVreling** [comment]
Maybe a good idea that all of this builds on [AEGEE/network#2](https://github.com/AEGEE/network/pull/2)

File: src/styles/main.scss [hunk 1]
```diff
@@ -248,6 +248,11 @@ h1, h2, h3, h4, h5, h6 {
   animation: rainbow-text 2s infinite;
 }
 
+.required:after {
```
Comment (WikiRik) [inline src/styles/main.scss:251]: Wait, would this add a red star to every required thing? I would put this at a different PR since this impacts the entire frontend

File: src/views/core/bodies/ChangeBoardModal.vue [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+        president: this.board.positions[0].id,
+        secretary: this.board.positions[1].id,
+        treasurer: this.board.positions[2].id,
+        // TODO: Think of a good way to export other positions
+        // Probably just with function and id
+        // other_members: {
+        //
+        // },
+        message: this.board.message
+        // TODO: Add image
```
Comment (WikiRik) [inline src/views/core/bodies/ChangeBoardModal.vue:149]: If this works, maybe we could do the same at event creation?

File: src/styles/main.scss (hunk 1 above)
Comment (LeonVreling) [reply src/styles/main.scss:251]: Ah oops did not mean to push that yet, since it only works for like half of the cases 😅

File: src/views/core/bodies/ChangeBoardModal.vue (hunk 2 above)
Comment (LeonVreling) [reply src/views/core/bodies/ChangeBoardModal.vue:149]: You mean with the image uploading and stuff?

File: src/views/network/BoardListing.vue [hunk 3]
```diff
@@ -115,10 +114,11 @@ export default {
         { value: 'other', name: 'Other' }
       ],
       isLoading: false,
+      source: null,
```
Comment (LeonVreling) [inline src/views/network/BoardListing.vue:117]: I don't think the cancel token is working correctly, and adding this line did not fix it

Comment (WikiRik) [reply src/views/network/BoardListing.vue:117]: Does this info help you? https://github.com/axios/axios#cancellation

File: src/views/core/bodies/ChangeBoardModal.vue (hunk 2 above)
Comment (WikiRik) [reply src/views/core/bodies/ChangeBoardModal.vue:179]: Yes

File: src/views/network/BoardListing.vue (hunk 3 above)
Comment (LeonVreling) [reply src/views/network/BoardListing.vue:117]: Ah it should be fixed with the newest commit

File: src/router/routes.js [hunk 4]
```diff
@@ -262,6 +262,14 @@ module.exports = [
       auth: false
     }
   },
+  {
+    name: 'oms.bodies.boards',
+    path: '/bodies/:id/boards',
+    component: 'core/bodies/ViewBoards',
```
Comment (WikiRik) [inline src/router/routes.js:268]: Why is this not part of network?

File: src/views/core/bodies/Single.vue [hunk 5]
```diff
@@ -345,9 +407,40 @@ export default {
...
+
+          for (const position of board.other_members) {
+            this.$set(position, 'user', users.find(user => user.id === position.user_id))
+          }
+        }
+      }).catch((err) => {
+        if (err.response.status === 404) {
+          return
+        }
+        // this.isLoading = false
```
Comment (WikiRik) [inline src/views/core/bodies/Single.vue:440]: Why is this still here?

**Review by WikiRik — COMMENTED** [review]
In general looks good already, nice work!

File: src/router/routes.js (hunk 4 above)
Comment (WikiRik) [reply src/router/routes.js:268]: Apparently it's easier this way due to the ChangeBoardModal component and the structure in the folders.

**Comment by WikiRik** [comment]
@serge1peshcoff this is working now so after the final tests of Network Director/NetCom we can merge this and publish the first thing of the network module

**Comment by WikiRik** [comment]
This is of course merged with https://github.com/AEGEE/network/pull/2 and https://github.com/AEGEE/mailer/pull/65

**Comment by LeonVreling** [comment]
@WikiRik do you maybe know what the status of this PR is?

**Comment by WikiRik** [comment]
We need to rebase and retest but I think this is ready to be merged apart from that

**Review by WikiRik — APPROVED** [review]
We can merge this in, without the manage_network:boards or view:board permissions nobody should see this
